src/core/mac/mac_frame.cpp

```diff
@@ -193,6 +199,34 @@ uint8_t Frame::FindDstPanIdIndex(void) const
     return index;
 }
 
+bool Frame::IsDstPanIdPresent(uint16_t aFcf) const
+{
+    bool present          = true;
+    bool isDstAddrPresent = IsDstAddrPresent(aFcf);
+
+    if ((aFcf & kFcfFrameVersionMask) == kFcfFrameVersion2015)
+    {
+        bool isPanIdCompressed = aFcf & kFcfPanidCompression;
+        bool isSrcAddrPresent  = IsSrcAddrPresent(aFcf);
+        // According to IEEE 802.15.4-2015 Table 7-2
+        if ((!isDstAddrPresent && !isSrcAddrPresent && !isPanIdCompressed) ||
+            (isDstAddrPresent && !isSrcAddrPresent && isPanIdCompressed) ||
+            (!isDstAddrPresent && isSrcAddrPresent && isPanIdCompressed) ||
+            (!isDstAddrPresent && isSrcAddrPresent && !isPanIdCompressed) ||
```

Comment:
```suggestion
            (!isDstAddrPresent && isSrcAddrPresent) ||
```

Reply:
This is actually to make the code easier to understand according to the table. I will try another implementation using `switch`.

Reply:
I made another implementation using `switch`:
```
    if (IsVersion2015(aFcf))
    {
        switch (aFcf & (kFcfDstAddrMask | kFcfSrcAddrMask | kFcfPanidCompression))
        {
        case (kFcfDstAddrNone | kFcfSrcAddrNone):
        case (kFcfDstAddrExt | kFcfSrcAddrNone | kFcfPanidCompression):
        case (kFcfDstAddrShort | kFcfSrcAddrNone | kFcfPanidCompression):
        case (kFcfDstAddrNone | kFcfSrcAddrExt):
        case (kFcfDstAddrNone | kFcfSrcAddrShort):
        case (kFcfDstAddrNone | kFcfSrcAddrExt | kFcfPanidCompression):
        case (kFcfDstAddrNone | kFcfSrcAddrShort | kFcfPanidCompression):
        case (kFcfDstAddrExt | kFcfSrcAddrExt | kFcfPanidCompression):
            present = false;
            break;
        default:
            break;
        }
    }
```
What do you think? @simonlingoogle @librasungirl @bukepo @brianljt @abtink

Reply:
LGTM.